### PR TITLE
Removes lerna and joins sub packages

### LINKED ISSUE #443: Remove lerna and join packages
We need to refactor the basic setup of this repo.
We agreed to 
1.) remove lerna
2.) join the three different packages into one (baseclient)
3.) Eventually refresh the project setup without the clutter of create-react-app after ejecting

This should enhance the usability and simplicity of this project.

### PR DESCRIPTION
**Pull request by KaiVolland**
This PR removes `lerna` and adapts the structure of the project.

- Contains some adaptions to TypeScript.
  - Removes `ts-jest`
  - Adapts `index.d.ts`
- Sets the node versions in CI to 10 and 12.
- Replaces `canvas-prebuilt` with `canvas` package
- Imports modules with direct paths instead of import form their index files

Closes #439 
Closes #443

### CODE REVIEW

**Comment by weskamm** [comment]
As long as the (puppeteer) tests go green, :+1: from me

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 node_js:
-- "8"
```
Comment (dnlkoch) [inline .travis.yml]: Is the support for v8 gone with the proposed changes?

File: README.md [hunk 2]
```diff
@@ -12,9 +12,6 @@ and then
 
 `npm i`
 
-This will automatically install package dependencies with `lerna bootstrap`.
-It will also automatically compile the submodules like react-geo, the state and components module via postinstall.
-
```
Comment (dnlkoch) [inline README.md]: There are some other `lerna` related changes needed in the README, see e.g. chapter `How to add a dependency?`. Please also check the current state of the listed `TODOs`.

File: config/jest/jest.config.js [hunk 3]
```diff
@@ -12,23 +13,26 @@ module.exports = {
...
   transformIgnorePatterns: [
-    './../../baseclient-components/',
     '/node_modules/(?!(ol|css-animation|labelgun|mapbox-to-ol-style|ol-mapbox-style|antd|@terrestris|(rc-*[a-z]*))/).*/'
   ],
   setupFiles: [
-    '<rootDir>/../config/jest/spec/setup.js'
+    '<rootDir>/config/jest/spec/setup.js'
+  ],
+  setupFilesAfterEnv: [
+    "jest-canvas-mock"
```
Comment (dnlkoch) [inline config/jest/jest.config.js]: Minor: Please unify usage of single and double quotes.

File: config/webpack.common.config.js [hunk 4]
```diff
@@ -132,21 +132,11 @@ const commonWebpackConfig = {
...
-      "baseclient-state": path.join(__dirname + '/../../', 'baseclient-state'),
-      ol: path.join(__dirname + '/../', 'node_modules', 'ol'),
-      react: path.join(__dirname + '/../', 'node_modules', 'react'),
-      "react-redux": path.join(__dirname + '/../', 'node_modules', 'react-redux'),
-      "react-i18next": path.join(__dirname + '/../', 'node_modules', 'react-i18next'),
-      tslib: path.join(__dirname + '/../', 'node_modules', 'tslib'),
-    },
-    modules: ['node_modules', '../baseclient/node_modules', './packages/baseclient/node_modules']
+      // This is need if working with npm link
+      'ol': path.join(__dirname, '../node_modules', 'ol')
```
Comment (dnlkoch) [inline config/webpack.common.config.js]: Please comment this line as it should only needed when working with a linked `@terrestris/react-geo` package.

This should then added as a note the the README (under section `Development`).

File: package.json [hunk 5]
```diff
@@ -37,19 +37,133 @@
...
-    "bootstrap": "lerna bootstrap --no-ci",
-    "test": "lerna run test --scope *baseclient*",
-    "start-static": "lerna run start-static --stream",
-    "start-shogun2": "lerna run start-shogun2 --stream",
-    "build:static": "cd packages/baseclient/ && npm run build:static && cd ../../",
-    "build:shogun2": "cd packages/baseclient/ && npm run build:shogun2 && cd ../../",
-    "build:ts": "lerna run build:ts --stream",
-    "postinstall": "npm run bootstrap && cd packages/react-geo && npm run build:dist && cd ../baseclient-state && npm run build:ts && cd ../baseclient-components && npm run build:ts"
+    "start:static": "webpack-dev-server --config config/webpack.static.config.js --hot",
+    "start:shogun2": "env-cmd .shogunrc webpack-dev-server --config config/webpack.shogun2.config.js --hot",
```
Comment (dnlkoch) [inline package.json]: Is this successfully tested?

File: src/util/PrintUtil/PrintUtil.spec.ts [hunk 6]
```diff
@@ -1,42 +1,43 @@
 /*eslint-env jest*/
 import OlImageWmsSource from 'ol/source/ImageWMS';
-import OlVectorSource from 'ol/source/Vector';
-import OlVectorLayer from 'ol/layer/Vector';
 import OlImageLayer from 'ol/layer/Image';
 
-import TestUtils from '../../../spec/TestUtils';
+// This causes issues via rbush depency
```
Comment (dnlkoch) [inline src/util/PrintUtil/PrintUtil.spec.ts]: Please note this as an issue.

File: .travis.yml (hunk 1 above)
Comment (KaiVolland) [reply .travis.yml]: Nope. We could test all 3 versions. Should i go for it?

Comment (marcjansen) [reply .travis.yml]: 10 plus is enough IMO

Comment (dnlkoch) [reply .travis.yml]: 10 is absolutely sufficient, but could you please add an engine block to the `package.json` with the supported versions?

File: package.json (hunk 5 above)
Comment (KaiVolland) [reply package.json]: No. Not yet. I think i'll tackle this with @mholthausen on monday. There might be a follow up PR incoming.

File: src/util/PrintUtil/PrintUtil.spec.ts (hunk 6 above)
Comment (KaiVolland) [reply src/util/PrintUtil/PrintUtil.spec.ts]: https://github.com/terrestris/react-geo-baseclient/issues/449

File: .travis.yml (hunk 1 above)
Comment (KaiVolland) [reply .travis.yml]: https://github.com/terrestris/react-geo-baseclient/pull/450

**Comment by dnlkoch** [comment]
Please recheck the git history…

**Pull request merged**